[MDEV-30366] Bulk unitary result flag client implementation part.

rusher opened the pull request:
With MDEV-30366, server now permit to send a result-set containing generated id and Affected rows for each bulk operation. This feature can be enabled with option MARIADB_OPT_BULK_UNIT_RESULTS when server supports it.

9EOR9 commented on libmariadb/mariadb_lib.c:
```diff
@@ -3830,6 +3832,9 @@ mysql_optionsv(MYSQL *mysql,enum mysql_option option, ...)
     }
     break;
+  case MARIADB_OPT_BULK_UNIT_RESULTS:
+    mysql->options.bulk_unit_results= *(my_bool *)arg1;
```
This will break ABI, since the size of MYSQL will change. Intstead option should be stored in mysql->options.extensions via OPT_SET_EXTENDED_VALUE (and bulk_unit_results should be moved from st_mysql_options to st_mysql_options_extension in ma_common.h)

9EOR9 commented on unittest/libmariadb/bulk1.c:
```diff
@@ -21,6 +21,13 @@
 
 static my_bool bulk_enabled= 0;
 
+#define SERVER_SUPPORT_BULK_UNIT_RESULTS(stmt)\
```
better to pass mysql instead of stmt to the macro, so it could be used without creating a statement, e.g. to skip a test .

9EOR9 commented on unittest/libmariadb/bulk1.c:
```diff
@@ -1067,6 +1077,345 @@ static int test_mdev16593(MYSQL *mysql)
+  unsigned int intval;
+
+  SKIP_MAXSCALE;
+
```
```
if (!SERVER_SUPPORT_BULK_UNIT_RESULTS(my))
{
  diag("Server doesn't support bulk unit results");
  return SKIP;
}
```

9EOR9 reviewed (commented):
review finished (see comments above)

rusher committed:
[MDEV-30366] Bulk unitary result flag client implementation part.

With MDEV-30366, server now permit to send a result-set containing generated id and Affected rows for each bulk operation. This feature can be enabled with option MARIADB_OPT_BULK_UNIT_RESULTS when server supports it.
